ansible_collections/arista/avd/roles/eos_designs/templates/network_services/logic.j2

```diff
@@ -38,6 +38,21 @@
 {%                     for svi in vrf.svis | arista.avd.convert_dicts('id') | arista.avd.natural_sort('id') %}
 {%                         if svi.id | int in whitelist.vlans %}
 {%                             if "all" in match_tags or svi.tags | arista.avd.default(['all']) is arista.avd.contains(match_tags) %}
+{%                                 if svi.profile is arista.avd.defined %}
+{%                                     set svi_profile = svi_profiles | arista.avd.default({}) |
+                                                                        arista.avd.convert_dicts('profile') |
+                                                                        selectattr('profile', 'arista.avd.defined', svi.profile) |
+                                                                        first %}
+{%                                     if svi_profile.parent_profile is arista.avd.defined %}
+{%                                         set svi_parent_profile = svi_profiles | arista.avd.default({}) |
+                                                                                   arista.avd.convert_dicts('profile') |
+                                                                                   selectattr('profile', 'arista.avd.defined', svi_profile.parent_profile) |
+                                                                                   first %}
+{%                                     endif %}
+{%                                     set svi = {} | ansible.builtin.combine(svi_parent_profile | arista.avd.default({}),
+                                                      svi_profile | arista.avd.default({}),
+                                                      svi, recursive=true, list_merge='replace') %}
+{%                                 endif %}
 {%                                 do tmp_svis.append(svi) %}
```

Comment:
We have to set structured_config here, to avoid loosing the originial `svi.structured_config`, since the recursive merge would overwrite it.
```suggestion
{%                                     set tmp_svi = {} | ansible.builtin.combine(svi_parent_profile | arista.avd.default({}),
                                                      svi_profile | arista.avd.default({}),
                                                      svi, recursive=true, list_merge='replace') %}
{#                                     Structured config should not be merged recursively, but will be taken directly from the most specific level #}
{%                                     do tmp_svi.update({"structured_config": svi.nodes[inventory_hostname].structured_config | arista.avd.default(
                                            
         svi_profile.nodes[inventory_hostname].structured_config,
                                            
         parent_profile.nodes[inventory_hostname].structured_config,
                                                    svi.structured_config,
                                                    svi_profile.structured_config,
                                                    parent_profile.structured_config)}) %}
{%                                 endif %}
{%                                 do tmp_svis.append(tmp_svi) %}
```